rholang/src/test/scala/coop/rchain/rholang/interpreter/ReduceSpec.scala

```diff
@@ -2260,133 +2166,130 @@ class ReduceSpec extends FlatSpec with Matchers with PersistentStoreTester {
           EList(
             List[Par](
+              GString("a"),
+              ETupleBody(ETuple(Seq(GString("b"), GInt(2L))))
             )
           )
         ),
-        List()
-      )
-
-    val result: Par = withTestSpace(errorLog) {
-      case TestFixture(_, reducer) =>
-        implicit val env: Env[Par] = Env[Par]()
-        val task = reducer.evalExpr(toMapCall)
-        Await.result(task.runToFuture, 3.seconds)
-    }
-    val resultList =
-      EMapBody(
-        ParMap(
-          List(
-          )
-        )
-      )
-    result.exprs should be(Seq(Expr(resultList)))
-    errorLog.readAndClearErrorVector().unsafeRunSync should be(Vector.empty[InterpreterError])
-  }
-
-  it should "return an error when `toMap` is called with arguments" in {
-    implicit val errorLog: ErrorLog[Task] = new ErrorLog[Task]()
-    val toMapCall: EMethod =
+        List[Par]()
+      ),
+      MethodNotDefined("toMap", "types except List[(K,V)]")
+    ),
+    ("""[("a", 1)].toMap(1) => MethodArgumentNumberMismatch""",
       EMethod(
         "toMap",
-        EMapBody(
-          ParMap(
-            List[(Par,Par)](
-              (GString("a"), GInt(1L))
+        EListBody(
+          EList(
+            List[Par](
+              ETupleBody(ETuple(Seq(GString("a"), GInt(1L))))
             )
           )
         ),
         List(GInt(1))
-      )
-
-    val result = withTestSpace(errorLog) {
-      case TestFixture(space, reducer) =>
-        implicit val env = Env[Par]()
-        val inspectTask  = reducer.eval(toMapCall) >> space.toMap
-        Await.result(inspectTask.runToFuture, 3.seconds)
-    }
-    result should be(mutable.HashMap.empty)
-    errorLog.readAndClearErrorVector().unsafeRunSync should be(
-      Vector(MethodArgumentNumberMismatch("toMap", 0, 1))
-    )
-  }
-
-  it should "return an error when `toMap` is called on GInt" in {
-    implicit val errorLog: ErrorLog[Task] = new ErrorLog[Task]()
-    val toMapCall: EMethod =
+      ),
+      MethodArgumentNumberMismatch("toMap", 0, 1)
+    )  ,
+    ("1.toMap() => MethodNotDefined",
       EMethod(
         "toMap",
         GInt(1L),
         List()
-      )
+      ),
+      MethodNotDefined("toMap", "Int")
+    ),
+    ("[].toSet(1) => MethodArgumentNumberMismatch",
+      EMethod(
+        "toSet",
+        ESetBody(
+          ParSet(
+            List(
+            )
+          )
+        ),
+        List(GInt(1))
+      ),
+      MethodArgumentNumberMismatch("toSet", 0, 1)
+    ),
+    ("1.toSet() => MethodNotDefined",
+      EMethod(
+        "toSet",
+        GInt(1L),
+        List()
+      ),
+      MethodNotDefined("toSet", "Int")
+    )
+  )
 
-    val result = withTestSpace(errorLog) {
-      case TestFixture(space, reducer) =>
-        implicit val env = Env[Par]()
-        val inspectTask  = reducer.eval(toMapCall) >> space.toMap
-        Await.result(inspectTask.runToFuture, 3.seconds)
+  "Reducer" should """return report errors in failure cases""" in {
```

Comment:
Nit: you don't need the extract quotations around `"return report errors in failure cases"`.